### ISSUE TITLE
as_array() with type casting: DB::query(Database::SELECT, '...')->execute()->as_array('id', array('id'=>'int'));

### ISSUE THREAD

**Issue by Daijobou** [koseven/koseven#504]
This is where the as_array() is assembled:
https://github.com/koseven/koseven/blob/master/modules/database/classes/Kohana/Database/Result.php#L98

What I miss here completely is the casting to a type, directly when creating as_array(). Too often you have to adjust the type in an extra loop, which would be unnecessary if you could specify it as a parameter in as_array().

Fictitious example:

$users = DB::query(Database::SELECT, 'SELECT id,username,email,score')->execute()->as_array('id');
$stats = DB::query(Database::SELECT, 'SELECT id,value')->execute()->as_array('id');
if (isset($stats[$users[5]) && $value>$users[5]['score']) echo 'yes';

Mixed values working but its sometimes tricky too. So I would like to be able to predefine ID as an integer and Score as a float and have that already provided in as_array(). Yes, you can casting directly in condition "$value>(float)$users[5]['score']" but you must do it each time or define before. 

Possible implementation as optional array-parameter:

$users = DB::query(Database::SELECT, 'SELECT id,username,email,score')->execute()->as_array('id', array('id'=>'int', 'score'=>'float'));

// field => data type
array('id'=>'int', 'score'=>'float')

So casting to "int", "float", "bool" would be great. Default is already "string".

Will probably never come, but I just wanted to post it here :)

**Comment by toitzi** [comment]
Yeah, that would be great, but not worth the effort right now, might be better to replace laravel orm with eloquent or so, will close this.
